main.go

```diff
@@ -13,29 +14,28 @@ func main() {
 		help()
 		return
 	}
-	var err error
-	switch flag.Arg(0) {
-	case "new":
-		err = commands.New()
-	case "login":
-		err = commands.Login(flag.Arg(1))
-	case "edit":
-		err = commands.Edit(flag.Arg(1))
-	case "worked":
-		err = commands.Worked(flag.Arg(1))
-	case "link":
-		err = commands.Link(flag.Arg(1))
-	case "editor":
-		err = commands.Editor(flag.Arg(1))
-	default:
-		err = commands.Query(flag.Arg(0))
-	}
-	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
+
+	if c, ok := commands.Commands[flag.Arg(0)]; !ok {
```

Comment:
Why is the query command any special? So far the main.go was really easy to understand - if a bit manual - now it's hard to get what's going on.